woocommerce/woocommerce-ios
# Hub Menu: Disable multiple taps on menu items

Closes: #6436 


### Description

This PR adds a workaround to disable multiple taps on the items of the Hub Menu.

The solution is:
- Keep a state in the hub menu for whether the menu items should be disabled temporarily
- Send the state as a binding property to each item - the items will have to update this value to false when its button receives a tap event.
- Reset the state to make sure all items are tappable after the view appears again / when modals are dismissed.

The downside of this solution is that there's a potential case when the tap state is not reset properly causing the menu items to be disabled. `onAppear` isn't called when modals are dismissed, so we need to take care of resetting the selection state on the dismissal of every modal. 

### Testing instructions

- Launch the app and navigate to the Menu tab.
- Try tapping on any item more than one time or tapping on more items at a time.
- Notice that only one new flow is presented at a time.

### Screenshots


https://user-images.githubusercontent.com/5533851/159624770-bcbcb8b3-f02f-499f-a155-9cb1c70f851b.mov


---
